src/poetry/core/spdx/license.py

```diff
@@ -1,15 +1,15 @@
 from __future__ import annotations
 
-from collections import namedtuple
+from typing import NamedTuple
 
 
-class License(namedtuple("License", "id name is_osi_approved is_deprecated")):
```

Comment:
I'd consider just making this a frozen dataclass instead.

Reply:
Would it be a zero-cost drop-in though? `NamedTuple` is compatible with standard `tuple`, afaik dataclasses are not. I could make it into a dataclass and go with more refactoring if needed. However, now that I think about it, I wonder if the work here shouldn't be separated into 2 PRs, one that only adds workflow and json update and another with module refactor. WDYT?

Reply:
I'd be fine with splitting this into two.